From edd783eca413a3b47b718a8efb7b62ef1dd942ca Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Mon, 13 May 2013 08:55:47 +0800
Subject: [PORT FROM MAIN]rework patch 76015 to avoid ANR if seek frequently in MR2

BZ: 48710

For the httplive streaming, the audio/video decoder may not be
setup at the same time. It will casue no video/audio output
for seeking frequently.
Original Patch:
37259: Seeking many times no video/audio output

Change-Id: I224e69ae2703cbc1e09d44b77bc3b699e501e7a7
Orig-Change-Id: Ibf468d3a5d1b8d137196583eb1edb8f30d3f75e3
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 .../nuplayer/HTTPLiveSource.cpp                    |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/HTTPLiveSource.cpp b/media/libmediaplayerservice/nuplayer/HTTPLiveSource.cpp
index c4e4d2f..574c136 100644
--- a/media/libmediaplayerservice/nuplayer/HTTPLiveSource.cpp
+++ b/media/libmediaplayerservice/nuplayer/HTTPLiveSource.cpp
@@ -100,6 +100,17 @@ sp<MetaData> NuPlayer::HTTPLiveSource::getFormatMeta(bool audio) {
         return NULL;
     }
 
+    // check if the format of the other stream is null,if it's still null
+    // return null for this stream's format to make sure two decoders can
+    // be initialized at the same time
+    ATSParser::SourceType otherType =
+        !audio ? ATSParser::AUDIO : ATSParser::VIDEO;
+    sp<AnotherPacketSource> anotherSource =
+        static_cast<AnotherPacketSource *>(mTSParser->getSource(otherType).get());
+    if (anotherSource != NULL && anotherSource->getFormat() == NULL) {
+        return NULL;
+    }
+
     return source->getFormat();
 }
 
-- 
1.7.4.1

